master/buildbot/test/fake/httpclientservice.py

```diff
@@ -113,7 +113,7 @@ def assertNoOutstanding(self):
         self.case.assertEqual(0, len(self._expected),
                               "expected more http requests:\n {!r}".format(self._expected))
 
-    def _doRequest(self, method, ep, params=None, data=None, json=None):
+    def _doRequest(self, method, ep, params=None, data=None, json=None, timeout=None):
```

Comment:
The goal of the fake is so that we can avoid to test the httpclientservuce users against treq and txrequests.

All the API that is on the fake is guaranteed to work on both implementation. It is a voluntarily subset of txrequests. If we start to add timeout, we start to break this guarantee